### PR TITLE
Normalize BQ table names in lineage

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

We lost the lineage because different treatment on BQ dataset name between BQ connector and dbt

### 🤓 What?

- Normalize the BQ name

### 🧪 Tested?

Add unit tests to cover

### CODE REVIEW

**Comment by mars-lan** [comment]
Why not normalize the lineage carwler instead?

File: metaphor/bigquery/lineage/extractor.py [hunk 1]
```diff
@@ -118,7 +120,7 @@ def _parse_view_lineage(self, project_id, bq_table: bigquery.table.Table) -> Non
             dataset_ids.add(
                 str(
                     to_dataset_entity_id(
-                        dataset_name.replace("`", ""),
+                        dataset_name.replace("`", "").lower(),
```
Comment (mars-lan) [inline metaphor/bigquery/lineage/extractor.py:123]: User `dataset_normalized_name` directly?

Comment (elic-eon) [reply metaphor/bigquery/lineage/extractor.py]: You're right. Does the replace back quote logic need to move to `dataset_normalized_name`?

Comment (mars-lan) [reply metaphor/bigquery/lineage/extractor.py]: Yes, make senses to move the quote stripping into `dataset_normalized_name` as well.

Comment (elic-eon) [reply metaphor/bigquery/lineage/extractor.py]: Updated

File: metaphor/common/entity_id.py [hunk 2]
```diff
@@ -78,7 +78,19 @@ def to_person_entity_id(email: str) -> EntityId:
 
 
 def dataset_normalized_name(
-    db: Optional[str] = None, schema: Optional[str] = None, table: Optional[str] = None
+    db: Optional[str] = None,
+    schema: Optional[str] = None,
+    table: Optional[str] = None,
+    lower_case: bool = True,
```
Comment (mars-lan) [inline metaphor/common/entity_id.py:84]: Would rather hard-code this to avoid any discrepancy when normalizing dataset names.

Comment (elic-eon) [reply metaphor/common/entity_id.py]: Make sense, updated.

File: metaphor/common/entity_id.py [hunk 3]
```diff
@@ -78,7 +78,16 @@ def to_person_entity_id(email: str) -> EntityId:
 
 
 def dataset_normalized_name(
-    db: Optional[str] = None, schema: Optional[str] = None, table: Optional[str] = None
+    db: Optional[str] = None,
+    schema: Optional[str] = None,
+    table: Optional[str] = None,
+    strip_backquote: bool = True,
```
Comment (mars-lan) [inline metaphor/common/entity_id.py:84]: This as well

Comment (elic-eon) [reply metaphor/common/entity_id.py]: Removed.

**Review by mars-lan — APPROVED** [review]

**Pull request merged**